common/display/displayplanemanager.cpp

```diff
@@ -197,42 +196,71 @@ std::tuple<bool, DisplayPlaneStateList> DisplayPlaneManager::ValidateLayers(
     }
   }
 
-  if (layer_begin != layer_end) {
     // Handle layers for overlay
+  DisplayPlaneState &last_plane = composition.back();
     uint32_t index = 0;
+    uint32_t layers_size = layers->size();
+
+    int32_t num_planes_compose = 0;
+
+    ALOGE("layers_size = %u\n", layers_size);
+    if (layers_size >= 4) {
+      num_planes_compose = layers_size - 3;
```

Comment:
no allocating the top layers to the overlays and then compositing & displaying remaining layers into the Primary plane.
Will remove the hardcode value